metricflow/query/query_parser.py

```diff
@@ -754,6 +758,25 @@ def _parse_linkable_elements(
             entity_specs=tuple(entity_specs),
         )
 
+    def _verify_resolved_granularity_for_date_part(
+        self,
+        requested_dimension_structured_name: StructuredLinkableSpecName,
+        partial_time_dimension_spec: PartialTimeDimensionSpec,
+        metric_references: Sequence[MetricReference],
+    ) -> None:
+        resolved_granularity = list(
+            self._time_granularity_solver.resolve_granularity_for_partial_time_dimension_specs(
+                metric_references=metric_references,
+                partial_time_dimension_specs=(partial_time_dimension_spec,),
+            ).values()
+        )[0].time_granularity
```

Comment:
Instead of doing the list(dict.values())[0], can we just use the index? The dict is keyed on the partial time dimension spec, and any spec that doesn't resolve will raise, so we should be able to do this:

```
resolved_granularity = self._time_granularity_solver.resolve_granularity_for_partial_time_dimension_specs(
    ...
)[partial_time_dimension_spec]
```

Reply:
Even cleaner would be to split the TimeGranularitySolver method into two - one that does a dict for multiple specs and one called something like `find_minimum_granularity_for_partial_time_dimension_spec` that returns a granularity value.

But I think it's totally reasonable to not do that in this PR and just use the dict access instead.

Reply:
Happy to do the latter!